src/System.IO.FileSystem/src/System/IO/ReadLinesIterator.cs

```diff
@@ -97,7 +97,7 @@ internal static ReadLinesIterator CreateIterator(string path, Encoding encoding)
 
         private static ReadLinesIterator CreateIterator(string path, Encoding encoding, StreamReader reader)
         {
-            return new ReadLinesIterator(path, encoding, reader ?? new StreamReader(FileStream.InternalOpen(path, useAsync: false), encoding));
+            return new ReadLinesIterator(path, encoding, reader ?? new StreamReader(new FileStream(path, FileMode.Open, FileAccess.Read, FileShare.Read), encoding));
```

Comment:
I'll deal with simplifying this where possible in a subsequent change. There is additional cleanup that can potentially be done with SR/W because FS is moving down.